### DISCUSSION TITLE
Trip Kit Update erforderlich?

### DISCUSSION THREAD

**Discussion by DeltaTango69** [mipastgt/AIPBrowserDE-Public#24] [Ideas]
Hallo Michael,

vielen Dank für Dein tolles Produkt! Ich nutze es schon seit längeren in Verbindung mit Enroute. Meistens erstelle ich mir ein Trip-Kit für ganz Deutschland - das ist ja jetzt mit der neueren Version ganz ohne Scipt sehr einfach möglich. Super!
Allerdings braucht der einige Zeit dafür und ich weiß nie wann es sich lohnt ein neues Trip-Kit zu erstellen? Ich hätte daher mal folgende fragen dazu:

- werden die AIP Daten immer neu eingelesen auch wenn sich nichts geändert hat?
- gibt es irgendwo eine Übersicht was sich seit der letzten Trip-Kit Erstellung geändert hat?

**Comment by mipastgt** [comment]
1. Bei der Erstellung eines Trip-Kits werden nur die Daten neu eingelesen die sich geändert haben. In Zeiten ausgedrückt sollte die erste vollständige Erzeugung eines Trip-Kits von ganz Deutschland bei einer guten Internetverbindung ca. 25 Minuten dauern. Jede weitere Erzeugung sollte dann ca. 2-3 Minuten dauern. Je nach den aktuellen Randbedingungen schwankt das natürlich immer etwas.
2. Eine Übersicht der aktuellen Änderungen gibt es derzeit tatsächlich nicht. Bei Bedarf könnte man das jedoch machen. Für mich selbst erstelle ich so eine Übersicht um zu wissen welche Georeferenzen ich aktualisieren muss. Bei der letzten Aktualisierung vor 14 Tagen waren das 161 Seiten und 51 georeferenzierte Charts. Ich habe die beiden Dateien mal zur Information angehängt.

Die Frage wäre nun welche der Information für den Endbenutzer, also Dich, von Interesse wären und in welcher Form. Listen zu erzeugen, die dann doch keiner anschaut, wäre natürlich Zeitverschwendung. Wegen der Menge der Änderungen sollte man idealerweise schon alle 28 Tage ein Update des Trip-Kits machen. Das Durchsuchen von Listen dauert vermutlich länger als die Erzeugung eines neuen Trip-Kits.

[aip-de-vfr-core-model-metadata-diff-orig.csv](https://github.com/user-attachments/files/17236264/aip-de-vfr-core-model-metadata-diff-orig.csv)
[aip-de-vfr-geo-info-updates-required-orig.csv](https://github.com/user-attachments/files/17236267/aip-de-vfr-geo-info-updates-required-orig.csv)

**Comment by DeltaTango69** [comment]
Für mich wäre die zweite Liste schon interessant. In dem Trip Kit für enroute interessieren ja auch nur die Charts. Vielleicht gibt es ja irgendwann mal ein Update so das auch die Flugplatzinformationen noch interessant wären. Aber wenn es einfach möglich wäre die beiden Listen mit auszugehen - es reicht ja die einfach mit ins Verzeichnis zu schreiben - würde ich das schon gut finden.

**Reply by mipastgt** [reply]
Ich schau mal was sich machen läßt.

**Comment by mipastgt** [comment]
Ich plane das in folgender Weise zu implementieren:
In der Applikation kann der Benutzer ein Datum und einen Filter (AllPages, AerodromePages, ChartPages) auswählen. Auf Knopfdruck wird dann eine CSV-Datei mit allen geänderten Seiten entsprechend der obigen Auswahl erstellt. Im Anhang findet sich eine ZIP-Datei mit allen drei Varianten der Änderungen seit dem 5. Sept. 2024.
[modified-pages-csv-files.zip](https://github.com/user-attachments/files/17277078/modified-pages-csv-files.zip)

**Comment by DeltaTango69** [comment]
Hallo Michael,

ja das sieht super aus! Würde mich über die Umsetzung freuen - vielen Dank!

Gruß

Dirk

**Comment by mipastgt** [comment]
Das Feature ist jetzt implementiert und kann in der aktuellen [Preview](https://mpmediasoft.de/products/AIPBrowserDE/help/AIPBrowserDE-Preview.html) des **AIP Browser DE** getested werden.

**Comment by mipastgt** [comment]
Man erreicht den Dialog über die Einstellungen.
<img width="511" alt="Bildschirmfoto 2024-10-09 um 11 26 18" src="https://github.com/user-attachments/assets/c97f2402-2b54-46a2-96d4-120171d3e999">
Die Liste wird wie üblich im Ausgabeordner gespeichert.

**Comment by DeltaTango69** [comment]
Hallo Michael,

vielen Dankfür die scnelle Umsetzung. Die Dateiausgabe erfolgt bei mir - sehr schön. Aber mit der neuen Version bekommeich jetzt kein Trip-Kit mehr erstellt. 

Nach "The selected trip kit data is now exported. This may take a while. Please be patient." passiert nicht mehr. Im Ausgabevezeichniss wird nur ein leeres Verzeichniss "charts" angelegt.

Gruß Dirk

**Comment by mipastgt** [comment]
Kannst Du mir die Log-Datei `.AIPBrowserDE/logs/AIPBrowserDE.log` aus dem versteckten Ordner `.AIPBrowserDE` schicken?

**Comment by DeltaTango69** [comment]
...

**Reply by mipastgt** [reply]
Ich sehe keine Log-Datei. Am einfachsten ist es wenn Du die zipst und dann einfach hier an eine Message hängst. Das Mail-Interface hier scheint entweder keine Anhänge zu können oder akzeptiert keine unkomprimierten Dateien. Oder einfach als Mailj-Anhang direkt an mich.

**Comment by DeltaTango69** [comment]
Hallo Michael, hier noch mal als ZIP. Das log gibt aber auch nicht viel her, befürchte ich.
Gruß Dirk

**Comment by mipastgt** [comment]
Sorry, aber hier hängt auch keine ZIP-Datei dran. Schick's mir doch einfach per E-Mail an `AIPBrowserDE@mpMediaSoft.de`

**Comment by mipastgt** [comment]
Danke für die Log-Datei. Sie sagt tatsächlich nicht sehr viel. Inzwischen kann ich den Fehler auch selbst reproduzieren, indem ich auf Windows teste und nicht auf dem Mac. Der Fehler selbst produziert keine Fehlermeldung und das Programm funktionierts auch danach ganz normal. Es sieht so aus als ob in dem gewählten Gebiet einfach keine Charts gefunden werden und deshalb ein leeres Trip-Kit erzeugt wird. Bisher habe ich keine Idee woran das liegen könnte und warum der Fehler auf dem Mac nicht auftritt.

**Comment by DeltaTango69** [comment]
Ok, vorher hat das ja ohne Probleme funktioniert?!

**Comment by mipastgt** [comment]
Ok, eine neue Version liegt jetzt auf dem Server und kann in der aktuellen [Preview](https://mpmediasoft.de/products/AIPBrowserDE/help/AIPBrowserDE-Preview.html) des AIP Browser DE getested werden. Der Fehler ist tricky und hat auch nichts mit der hier diskutierten Erweiterung zu tun. Ein Code-Shrinking Tool funktioniert nicht mehr richtig und ich musste das vorläufig abschalten. Darum sind die neuen Installer jetzt auch etwas größer als vorher. Siehe: https://github.com/Guardsquare/proguard/issues/441
